### PR TITLE
Add Zilliz Cloud Pipeline as a Managed Index

### PR DESCRIPTION
**Pull request by jaelgu**
Add a managed index powered by [Zilliz Cloud Pipeline](https://docs.zilliz.com/docs/pipelines)

- New feature (non-breaking change which adds functionality)
- Added new notebook (that tests end-to-end): `docs/examples/managed/zcpDemo.ipynb`

### CODE REVIEW

**Comment by hatianzhang** [comment]
@jaelgu awesome work. could you help fix some lint issues

**Comment by jaelgu** [comment]
@hatianzhang lint issues fixed. thank you!

File: docs/examples/managed/zcpDemo.ipynb [hunk 1]
```diff
@@ -0,0 +1,199 @@
+{
```
Comment (hatianzhang) [inline docs/examples/managed/zcpDemo.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.        metadata={"version": "milvus 2.3", "year": "2023"},  # optional
</pre><p>could you show an example which enable metadata filter as well</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/9605/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (jaelgu) [reply docs/examples/managed/zcpDemo.ipynb:1]: I've added example to apply metadata filters in as_query_engine

File: llama_index/indices/managed/zilliz/retriever.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        output_metadata: list = [],
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> None:
+        self.search_top_k = search_top_k
+        if filters:
+            exprs = []
+            for fil in filters.filters:
+                expr = f"{fil.key} == '{fil.value}'"
+                exprs.append(expr)
+            self.filter = " && ".join(exprs)
```
Comment (hatianzhang) [inline llama_index/indices/managed/zilliz/retriever.py:35]: is this only supporting exact match

**Review by hatianzhang — APPROVED** [review]
overall lgtm

File: llama_index/indices/managed/zilliz/retriever.py (hunk 2 above)
Comment (jaelgu) [reply llama_index/indices/managed/zilliz/retriever.py:35]: I will submit another pr to support other operations.

File: llama_index/indices/managed/zilliz/base.py [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+    """
+
+    def __init__(
+        self,
+        cluster_id: str,
+        token: str,
+        cloud_region: str = "gcp-us-west1",
+        pipeline_ids: Optional[Dict] = None,
+        collection_name: str = "zcp_llamalection",
```
[the remark this answers is not in the document]
Comment (jaelgu) [reply llama_index/indices/managed/zilliz/base.py:70]: We encourage users to enter the cloud_region or collection_name.
- Cloud region is fixed for now but will support other values soon.
- If no pipeline_ids is given, then collection_name will be used to create the index.

File: llama_index/indices/managed/zilliz/base.py [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+        index_struct = ZillizCloudPipelineIndexStruct(
+            index_id="_".join([str(x) for x in pipeline_ids.values()]),
+            summary="Zilliz Cloud Pipeline Index",
+        )
+
+        super().__init__(
+            show_progress=show_progress, index_struct=index_struct, **kwargs
+        )
+
+        domain = f"https://controller.api.{cloud_region}.zillizcloud.com/v1/pipelines"
```
[the remark this answers is not in the document]
Comment (jaelgu) [reply llama_index/indices/managed/zilliz/base.py:101]: Other parts of this domain is fixed.

**Comment by logan-markewich** [comment]
@hatianzhang we should link this in the docs :) we have a managed index section

**Comment by hatianzhang** [comment]
@logan-markewich good point
I can add that

**Pull request merged**